### PR TITLE
Get rid of Apache Commons

### PR DESCRIPTION
**Pull request by ArnyminerZ**

### CODE REVIEW

**Review by rfc2822 — CHANGES_REQUESTED** [review]
Therte's still an import in MiscUtils

**Comment by rfc2822** [comment]
Nothing has changed, did you push?

**Comment by ArnyminerZ** [comment]
My bad 😅

**Review by sunkup — CHANGES_REQUESTED** [review]
Failing test :)

**Review by sunkup — COMMENTED** [review]
Looks good :+1:

File: lib/src/main/kotlin/at/bitfire/ical4android/Event.kt [hunk 1]
```diff
@@ -9,55 +9,89 @@ import at.bitfire.ical4android.util.DateUtils.isDateTime
...
-import java.util.*
+import java.util.LinkedList
+import java.util.UUID
 
-class Event : ICalendar() {
+data class Event(
+    override var uid: String? = null,
+    override var sequence: Int? = null,
+
+    /** list of CUAs which have edited the event since last sync */
```
Comment (sunkup) [inline lib/src/main/kotlin/at/bitfire/ical4android/Event.kt:57]: What does CUAs mean? Does UA stand for user agent? Calendar user agent?

**Review by sunkup — APPROVED** [review]

File: lib/src/main/kotlin/at/bitfire/ical4android/Event.kt (hunk 1 above)
Comment (rfc2822) [reply lib/src/main/kotlin/at/bitfire/ical4android/Event.kt:57]: Yes, as used in RFC 5545. But it's maybe better to write it out.

**Review by rfc2822 — CHANGES_REQUESTED** [review]

File: lib/src/main/kotlin/at/bitfire/ical4android/AndroidEvent.kt [hunk 2]
```diff
@@ -132,20 +131,22 @@ abstract class AndroidEvent(
...
+
+    /**
+     * Returns the full event data, either from [event] or, if [event] is null, by reading event
+     * number [id] from the Android calendar storage
+     * @throws IllegalArgumentException if event has not been saved yet
+     * @throws FileNotFoundException if there's no event with [id] in the calendar storage
+     * @throws RemoteException on calendar provider errors
+     */
+    val event: Event?
         get() {
```
Comment (rfc2822) [inline lib/src/main/kotlin/at/bitfire/ical4android/AndroidEvent.kt:147]: I think we can also add a `private set()` that sets the `_event`. Then we can use the private setter in the other methods (constructor, maybe somewhere else, …) and they don't have to know whether they shall write to `_event` or directly to `event`.

**Review by rfc2822 — CHANGES_REQUESTED** [review]

**Review by rfc2822 — APPROVED** [review]
Nice 😎

**Pull request merged**